pxr/imaging/plugin/hdRpr/volume.cpp

```diff
@@ -154,17 +159,59 @@ openvdb::CoordBBox computeGrigOnBB(TPtr grid)
 
 
 template <class TPtr, class TValueOnIt>
-void fillGridOnWithDefaultColor(TPtr grid, VtVec4fArray & out_grid, VtArray<size_t> & out_idx, openvdb::Coord & out_gridOnBBSize)
+void fillGridOnWithDefaultAlbedo(TPtr grid, VtArray<float> & out_density_grid, VtArray<size_t> & out_density_idx, VtArray<float> & out_albedo_grid, VtArray<size_t> & out_albedo_idx, openvdb::Coord & out_gridOnBBSize)
 {
-	openvdb::CoordBBox gridOnBB = computeGrigOnBB< TPtr, TValueOnIt>(grid);
+	openvdb::CoordBBox gridOnBB = computeGridOnBB< TPtr, TValueOnIt>(grid);
 	out_gridOnBBSize = gridOnBB.max() - gridOnBB.min();
 
 	for (TValueOnIt iter = grid->beginValueOn(); iter; ++iter) {
-		out_grid.push_back(defaultColor);
-		out_idx.push_back(coordToIndex(iter.getCoord(), gridOnBB, out_gridOnBBSize));
+		out_density_grid.push_back(grid->getAccessor().GetValue(iter.getCoord()));
+		out_albedo_grid.push_back(defaultColor);
+
+		size_t idx = coordToIndex(iter.getCoord(), gridOnBB, out_gridOnBBSize);
+		out_density_idx.push_back(idx);
+		out_albedo_idx.push_back(idx);
+	}
+}
+
+template <class TGrid>
+void ReadDensityFromGrid(openvdb::GridBase::Ptr inGrid, std::vector<uint32_t> &outDensityGridOnIndices, std::vector<float> &outDensityGridOnValueIndices, std::vector<float> &outDensityGridValues)
+{
+	TGrid::Ptr grid = openvdb::gridPtrCast<TGrid>(inGrid);
```

Comment:
type name dependent on `TGrid` requires `typename` specifier
[doc](https://en.cppreference.com/w/cpp/language/dependent_name)

Reply:
Removed template from this function in latest commit.